### ISSUE TITLE
Beta - New Upgrade System Issues

### ISSUE THREAD

**Issue by speed2CZ** [FAForever/fa#1057]
Factories: (Same for all types, all factions)
If you have T2 HQ then this is how upgrades available looks like for T1 factory. Arrow from T2 HQ to T2 Support Factory
![ss 2015-12-13 at 11 00 29](https://cloud.githubusercontent.com/assets/10947024/11766457/40cecfe6-a189-11e5-9b84-eae5311e1931.png)

Also no direct upgrade to T3 HQ from on T1 factory or T3 support factory on T1 factory
![ss 2015-12-13 at 11 06 18](https://cloud.githubusercontent.com/assets/10947024/11766464/8d82f7fe-a189-11e5-9813-6eaa029919a1.png)

Other Direct Upgrade Issues:
If you build T2 UEF Engy Station. Select it, upgade. Keeping it selected until upgrade is finished. Then you get the build option to build directly the upgaded one from the selected engy station. If you place it on map it will cancel since you cant really start building with engy station. Deselecting and selecting upgraded engy station will no longer show this build option.

Same can be done with UEF T2 shield, if you keep it selected, the shield has build option to builid another T3 shield, but also cancels the order if you try to place it on map

Final upgrade of Hives effected as well if upgraded directly.

This behaviour is also for Radars, in both cases, if upgraded step by step or directly. But it's not showing everytime, like 50% chance from my tests. Either it upgrades as it should and nothing to click in build menu anymore or you have option for final unit.

I've also noticed that building that have upgrade to T2/T3 like radar, hives, cybran shield. If you click direct upgrade to last stage and keeping the unit selected, the upgrades available are not always updating once one of the upgrades is done.

**Comment by aeoncleanse** [comment]
I've addressed this for 3652 in 05b0d7ca2233c34fe6a6a962206a769c5c4620b1

- For the confusion of arrows, I've simply turned them off for now.
- For the fact that selecting things allows you to almost order a build, it's strange and unwanted, but not dangerous to gameplay, so I think we can ignore it *for now*.
- For upgrade options not updating as upgrades complete, again it's not exactly harmful, so *for now* we'll leave it. This will also need changing in the future to be done properly.
